hack/update-config.sh

```diff
@@ -17,13 +17,21 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-TESTINFRA_ROOT=$(git rev-parse --show-toplevel)
+if [[ -n "${BUILD_WORKSPACE_DIRECTORY:-}" ]]; then
```

Comment:
This determines if we are running as part of a bazel build rule, right? It would be nice to have that in a comment.
Same with `TEST_WORKSPACE` below. (I'm assuming these are different because one rule is `sh_binary` and the other is `sh_test`?)